src/python/pants/backend/python/rules/run_setup_py.py

```diff
@@ -166,6 +167,33 @@ class SetupPySources:
     package_data: Tuple[PackageDatum, ...]
 
 
+@dataclass(frozen=True)
+class SetupPyKwargsRequest:
+    exported_target: ExportedTarget
+    requirements: ExportedTargetRequirements
+    sources: SetupPySources
+
+
+@frozen_after_init
+@dataclass(unsafe_hash=True)
+class SetupPyKwargs:
```

Comment:
I think this should be called `SetupKwargs`, as these are literally the keyword arguments to the `setup()` function call. The docstring should be modified appropriately. "Putting keyword arguments in a file" is an imprecise concept. "Calling a function with keyword arguments" is precise.

And `SetupKwargsRequest` likewise.